tests/specs/erc20/verification.k

```diff
@@ -25,6 +25,15 @@ module VERIFICATION-JAVA [symbolic, kast]
 endmodule
 
 module VERIFICATION-HASKELL [symbolic, kore]
-    import INFINITE-GAS
+    imports INFINITE-GAS
+    imports EDSL
+
+    rule #hashedLocation(_COMPILER, BASE, .IntList) => BASE [simplification]
 
+    rule keccakIntList(VS) => keccak(intList2ByteArray(VS)) [simplification]
+
+    rule intList2ByteArray(.IntList) => .ByteArray [simplification]
+    rule intList2ByteArray(V VS)     => #bufStrict(32, V) ++ intList2ByteArray(VS)
+      requires 0 <=Int V andBool V <Int pow256
+      [simplification]
```

Comment:
We probably don't want this solution (but it's definitely a valid one). The basic reason is that we have decided that the `keccakIntList` rule in `hashed-locations.md` should not be applied unless it has fully concrete arguments (we say this by adding that rule to the file `tests/specs/erc20/concrete-rules.txt`). So your second lemma here actually overrides that, and lowers us to the underlying definition of `keccakIntList`, rather than letting us reason about `keccakIntList` directly.

The issue is actually that the `hash1/hash2` abstraction here: https://github.com/kframework/evm-semantics/blob/master/tests/specs/lemmas.k#L166 is not being imported into the Haskell backend. So you could do this:

- Move the definitions of `hash1` and `hash2` to the file `hashed-locations.md`, and add in documentation about them.
- Move the lemmas about `keccakIntList` there into the common lemmas module `LEMMAS` in `tests/specs/lemmas.k`.
- Move the functional test you've added here to `tests/specs/functional/lemmas-spec.k`.

Reply:
Once you do that, you should confirm that this proof `hkg/allowance` still passes.

Reply:
Do we not want the `keccakIntList` rules in `hashed-locations.md` to stay there for the non-proof part of `kevm`?

Reply:
What I'd like to do is just mark all the `#hashedLocation` rules as `concrete`, and then remove everything about `keccakIntList` and such. Just write lemmas about `#hashedLocation` directly in the `hashed-locations.md` file.

Reply:
The new `#hashedLocation` abstraction actually _lowers_ everything to `keccak`, so you may need ot re-run the original proof to see what it's getting stuck on now, it's likely different than before.